app/src/providers/SessionProvider.tsx

```diff
@@ -49,7 +56,15 @@ export const SessionProvider: React.FC<SessionProviderProps> = ({ address, contr
     .then((isGovernor: boolean) => {
       const newUser = isGovernor ? DACS_DEFAULT : DEAL_DEFAULT;
       newUser.address = address;
-      setUser(newUser);
+      return newUser;
+    })
+    .then((user: User) => {
+      nameFromAddress(contracts.Ens, address)
+        .then((name: string) => {
+          user.name = name;
+          setUser(user);
+        })
+        .catch(console.log);
```

Comment:
Nested promises :(

Reply:
in this case you should of just returned nameFromAddress and continued on the promise chain normally